src/main/scala/uk/gov/nationalarchives/tdr/api/service/FFIDMetadataService.scala

```diff
@@ -74,26 +76,33 @@ class FFIDMetadataService(ffidMetadataRepository: FFIDMetadataRepository, matche
     }
   }
 
-  private def generateFileStatusRows(ffidMetadata: FFIDMetadataInput): List[FilestatusRow] = {
-    val uniqueStatuses: List[String] = ffidMetadata.matches.map(m => checkStatus(m.puid)).distinct
-
-    uniqueStatuses match {
-      case s if uniqueStatuses.size == 1 =>
-        List(FilestatusRow(uuidSource.uuid, ffidMetadata.fileId, FFID, s.head, Timestamp.from(timeSource.now)))
-      case _ => uniqueStatuses.filterNot(_.equals(Success)).map(
-        FilestatusRow(uuidSource.uuid, ffidMetadata.fileId, FFID, _, Timestamp.from(timeSource.now)))
-    }
+  private def generateFileStatusRows(ffidMetadata: FFIDMetadataInput): Future[List[FilestatusRow]] = {
+    val fileId = ffidMetadata.fileId
+    for {
+      consignments <- fileRepository.getConsignmentForFile(fileId)
+      consignmentType = if (consignments.isEmpty) { throw InputDataException(s"No consignment found for file $fileId") }
+        else { consignments.head.consignmenttype }
+      uniqueStatuses: List[String] = ffidMetadata.matches.map(m => checkStatus(m.puid, consignmentType)).distinct
+      rows = uniqueStatuses match {
+        case s if uniqueStatuses.size == 1 =>
+          List(FilestatusRow(uuidSource.uuid, ffidMetadata.fileId, FFID, s.head, Timestamp.from(timeSource.now)))
```

Comment:
Since there is a `fileId` declaration on row 80, I think that `ffidMetadata.fileId` on line 88 and 90 could be replaced with that. Also, I think the `uuidSource.uuid` and `Timestamp.from(timeSource.now)` could also be declared at the top of the method

Reply:
Ah, yes. Cool.